sdk/eventhub/Azure.Messaging.EventHubs/src/CreateBatchOptions.cs

```diff
@@ -40,6 +40,58 @@ public class CreateBatchOptions : SendOptions
             }
         }
 
+        /// <summary>
+        ///   Allows a hashing key to be provided for the batch of events, which instructs the Event Hubs
+        ///   service map this key to a specific partition but allowing the service to choose an arbitrary,
+        ///   partition for this batch of events and any other batches using the same partition hashing key.
+        ///
+        ///   The selection of a partition is stable for a given partition hashing key.  Should any other
+        ///   batches of events be sent using the same exact partition hashing key, the Event Hubs service will
+        ///   route them all to the same partition.
+        ///
+        ///   This should be specified only when there is a need to group events by partition, but there is
+        ///   flexibility into which partition they are routed. If ensuring that a batch of events is sent
+        ///   only to a specific partition, it is recommended that the identifier of the position be
+        ///   specified directly when sending the batch.
+        /// </summary>
+        ///
+        /// <value>
+        ///   If the producer wishes to influence the automatic routing of events to partitions, the partition
+        ///   hashing key to associate with the event or batch of events; otherwise, <c>null</c>.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   If the <see cref="CreateBatchOptions.PartitionKey" /> is specified, then no <see cref="CreateBatchOptions.PartitionId" />
+        ///   may be set when sending.
+        /// </remarks>
+        ///
+        public string PartitionKey { get; set; }
+
+        /// <summary>
+        ///   If specified, the sender is requesting that events be sent to this specific partition.  If the identifier is not
```

Comment:
In this line, we are using the term "sender", a few lines below, we use "producer".
I would suggest to stick to one :)

Reply:
Fair point; we've still got inconsistencies that we're rooting out form the naming change way back when.   Thanks!

Reply:
form? :)

Reply:
Typing is hard.    😛